src/tracking.c

```diff
@@ -383,7 +392,8 @@ void trackingInvalidateKeyRaw(client *c, char *key, size_t keylen, int bcast) {
             continue;
         }
 
-        sendTrackingMessage(target,key,keylen,0);
+        /* We need to use id as client may be freed */
+        trackingScheduleKey(id,keyobj);
```

Comment:
good idea

Reply:
@huangzhw please note the idea about `current_client` (other than the diff suggestion about the rename)

Reply:
Sorry. I'm still working on it. Just push simple things.

Reply:
If we use `current_client`, there is problem with `BLMOVE`.

```
lrange list2 0 -1
*2
$1
x
$1
x
blmove list1 list2 left left 0  # this was blocked
>2
$10
invalidate
*1
$5
list2
$1
a
```

If `blmove` is blocked and unblocked. Because `current_client` is not this client, invalidation message will be before command response.

Or we can modify `lmoveHandlePush`, but I'm not sure whether there are other problems.

Reply:
Good catch, was there a test that uncovered that? if not, let's add one. 

Regarding `lmoveHandlePush`, did you mean to reorder the calls so that the in validation is at the end? how about temporarily setting `currecnt_client` when processing a blocked command?

Reply:
We can temporarily setting `currecnt_client`. But it's a little trivial as we need add 4 places.

Reply:
It sounds to me like the right thing to do.. 
That's when we're actually processing the BLMOVE command, so current client should be set.

Reply:
+1 to let's use current_client and make its behavior clear. I didn't think it was correctly set everywhere, and it was nebulously defined.